### DISCUSSION TITLE
How would I create a test file for a skinning element?

### DISCUSSION THREAD

**Discussion by M0RGaming** [ppy/osu#15457] [Q&A] [answered]
I'm working on creating a skinning element for a UR counter, and I was wondering how would I go about setting up the tests for it? I'd presume that I would want to create a list of hit events and pass it onto the judgement, though I'm unsure on if I need to do anything special for it. I am also unsure of how to actually run the test. If possible, could you point me in the right direction?

Thank you for your time,
- M0R

**Comment by bdach** [comment, marked as answer]
just to confirm my understanding as to what you're trying to add: are you trying to add a numeric unstable rate counter?

if that is what you're trying to add, then please check `TestSceneHitErrorMeter` to see how that is done for existing HUD elements that use hit offset from judgement results

**Reply by peppy** [reply]
I'd also add that if you're adding a new UI element, start by making the display work in a single PR. Skinning should probably be added as a second step.

**Reply by M0RGaming** [reply]
> 
> 
> I'd also add that if you're adding a new UI element, start by making the display work in a single PR. Skinning should probably be added as a second step.

Ok, sounds good, I'll make a PR after I finish this test file then.

**Comment by M0RGaming** [comment]
> just to confirm my understanding as to what you're trying to add: are you trying to add a numeric unstable rate counter?

Yup, that's correct.

Thank you! I'll take a look at that.
